benchmarks/benchmarks/preprocessing.py

```diff
@@ -7,116 +7,131 @@
 
 from typing import TYPE_CHECKING
 
+from asv_runner.benchmarks.mark import skip_for_params
+
 import scanpy as sc
 from scanpy.preprocessing._utils import _get_mean_var
 
-from .utils import lung93k, pbmc3k, pbmc68k_reduced
+from .utils import get_dataset
 
 if TYPE_CHECKING:
     from anndata import AnnData
 
+    from .utils import Dataset
+
 
 adata: AnnData
+batch_key: str | None
 
 
-def setup(*params: str):
-    """Setup all tests.
+def setup(dataset: Dataset, *_):
+    """Setup global variables before each benchmark."""
+    global adata, batch_key
+    adata, batch_key = get_dataset(dataset)
 
-    The SparseDenseSuite below defines a parameter,
-    the other tests none.
-    """
-    global adata
 
-    if len(params) == 0 or params[0] == "pbmc68k_reduced":
-        adata = pbmc68k_reduced()
-    elif params[0] == "pbmc3k":
-        adata = pbmc3k()
-    elif params[0] == "lung93k":
-        adata = lung93k()
-    else:
-        raise ValueError(f"Unknown dataset {params[0]}")
+# The actual test suite begins here
 
+params: list[Dataset] = ["pbmc68k_reduced", "pbmc3k"]
+param_names = ["dataset"]
 
-def time_calculate_qc_metrics():
+
+def time_calculate_qc_metrics(*_):
     adata.var["mt"] = adata.var_names.str.startswith("MT-")
     sc.pp.calculate_qc_metrics(
         adata, qc_vars=["mt"], percent_top=None, log1p=False, inplace=True
     )
 
 
-def peakmem_calculate_qc_metrics():
+def peakmem_calculate_qc_metrics(*_):
     adata.var["mt"] = adata.var_names.str.startswith("MT-")
     sc.pp.calculate_qc_metrics(
         adata, qc_vars=["mt"], percent_top=None, log1p=False, inplace=True
     )
 
 
-def time_filter_cells():
-    sc.pp.filter_cells(adata, min_genes=200)
+def time_filter_cells(*_):
+    sc.pp.filter_cells(adata, min_genes=100)
 
 
-def peakmem_filter_cells():
-    sc.pp.filter_cells(adata, min_genes=200)
+def peakmem_filter_cells(*_):
+    sc.pp.filter_cells(adata, min_genes=100)
 
 
-def time_filter_genes():
+def time_filter_genes(*_):
     sc.pp.filter_genes(adata, min_cells=3)
 
 
-def peakmem_filter_genes():
+def peakmem_filter_genes(*_):
     sc.pp.filter_genes(adata, min_cells=3)
 
 
-def time_normalize_total():
+# scublet doesn’t work with these datasets
```

Comment:
why?

Reply:
Great question, the error is very obscure. I’ll see if I can find out. You suggested that it might have to do with ASV :thinking:

Reply:
Ah, I guess it’s just because that dataset comes preprocessed. scrublet wants raw counts.

Reply:
Ok so I didn't comment on this because I thought scanpy was supposed to be idempotent so I guess I'll bring it up now.  It seems like it might be a good idea to somehow mark what was done to which dataset in the functions (for example, I see `log1p` applied sometimes while others already have qc metrics calculated).  Do you think this is feasible?

It might worth at least renaming the datasets based on what has been done to them.

Reply:
Sorry, I guess in this case it's not a matter of idempotency.  The point still stands.

Reply:
Yeah, it would be great to document what processing has been done and to modernizing the datasets

1. right now: where possible without changing/removing stuff (e.g. add a `counts` layer to `pbmc68k_reduced`, as its `.raw` has the same number of genes and is therefore superflous)
2. for 2.0: remove old stuff, change things (e.g. call `make_obs_names_unique` before returning the data and remove `.raw`

Reply:
Amazing, thanks!  Do you think it's worth doing something here now, like somehow indicating [`pbmc3k` is log-transformed](https://github.com/scverse/scanpy/pull/3031/files#diff-f5f3918ea461b367ebfa4bca0592c40e7fd8bc074e4f8a3c373dabc7a32db0c4R36-R43) by the benchmark code specifically?

Reply:
I updated the whole PR to split the preprocessing benchmarks into before-transform and after-transform